plugins/store/kics/main.go

```diff
@@ -0,0 +1,272 @@
+		)
+		if err != nil {
+			log.Errorf("Failed to create KICS client: %v", err)
+			s.SetStatus(types.NewScannerStatus(types.Failed, types.Ptr(fmt.Errorf("failed to create KICS client: %w", err).Error())))
+			return
+		}
+
+		err = c.PerformScan(ctx)
+		if err != nil {
+			log.Errorf("Failed to perform KICS scan: %v", err)
+			s.SetStatus(types.NewScannerStatus(types.Failed, types.Ptr(fmt.Errorf("failed to perform KICS scan: %w", err).Error())))
+			return
+		}
+
+		if ctx.Err() != nil {
+			log.Errorf("The operation timed out: %v", ctx.Err())
+			s.SetStatus(types.NewScannerStatus(types.Failed, types.Ptr(fmt.Errorf("failed due to timeout %w", ctx.Err()).Error())))
+			return
+		}
+
+		err = s.formatOutput(outputFile, config.OutputFile)
+		if err != nil {
+			log.Errorf("Failed to format KICS output: %v", err)
+			s.SetStatus(types.NewScannerStatus(types.Failed, types.Ptr(fmt.Errorf("failed to format KICS output: %w", err).Error())))
+			return
+		}
+
+		log.Infof("Scanner finished running.")
+		s.SetStatus(types.NewScannerStatus(types.Done, types.Ptr("Scanner finished running.")))
+	}()
+}
+
+//nolint:gomnd
+func (s *KICSScanner) createScanParametersConfig(configPath *string) (*ScanParametersConfig, error) {
+	config := ScanParametersConfig{
+		PreviewLines:     3,
+		Platform:         []string{"Ansible", "CloudFormation", "Common", "Crossplane", "Dockerfile", "DockerCompose", "Knative", "Kubernetes", "OpenAPI", "Terraform", "AzureResourceManager", "GRPC", "GoogleDeploymentManager", "Buildah", "Pulumi", "ServerlessFW", "CICD"},
+		MaxFileSizeFlag:  100,
+		DisableSecrets:   true,
+		QueryExecTimeout: 60,
+		Silent:           true,
+		Minimal:          true,
+	}
+
+	if configPath == nil {
+		return &config, nil
+	}
+
+	file, err := os.Open(filepath.Clean(*configPath))
+	if err != nil {
+		return nil, fmt.Errorf("failed to open config file: %w", err)
+	}
+	defer file.Close()
+
+	ext := filepath.Ext(*configPath)
+
+	bytes, err := io.ReadAll(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read config file: %w", err)
+	}
+
+	switch ext {
+	case ".json":
+		if err := json.Unmarshal(bytes, &config); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal JSON config: %w", err)
+		} else {
+			return &config, nil
+		}
+
+	case ".yaml", ".yml":
+		if err := yaml.Unmarshal(bytes, &config); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal YAML config: %w", err)
+		} else {
+			return &config, nil
+		}
+
+	case ".toml":
+		if _, err := toml.Decode(string(bytes), &config); err != nil {
+			return nil, fmt.Errorf("failed to decode TOML config: %w", err)
+		} else {
+			return &config, nil
+		}
+
+	case ".hcl":
+		err := hclsimple.DecodeFile(*configPath, nil, &config)
+		if err != nil {
+			return nil, fmt.Errorf("failed to decode HCL config: %w", err)
+		} else {
+			return &config, nil
+		}
+
+	default:
+		return nil, fmt.Errorf("unsupported config file format: %s", ext)
+	}
```

Comment:
this is nice, we could move this logic to the `sdk` code so it could be reused by other plugins